### ISSUE TITLE
TechDocs: git feedback buttons not appearing automatically

### ISSUE THREAD

**Issue by leon-vg** [backstage/backstage#11026]
## Expected Behavior

A while ago, we automatically saw the feedback buttons appear for all TechDocs sites. However, now they don't show automatically.

![image](https://user-images.githubusercontent.com/14820561/164709429-4666e5b7-9a57-4ffa-9af3-cfe19526c459.png)

## Actual Behavior

The feedback buttons are gone. However, when you add a `repo_url` entry in mkdocs.yml, they are back.

Stepping through the code using breakpoints, I found that the [getRepoUrlFromLocationAnnotation](https://github.com/backstage/backstage/blob/c938e0d61dbe74f126e5e347413addba8bcd3698/plugins/techdocs-node/src/stages/generate/helpers.ts#L94) helper function only adds the `repo_url` OR `edit_uri` properties. It seems that to show the feedback buttons both properties are necessary (?).

## Steps to Reproduce

1. Have a TechDocs repository without `repo_url` and `edit_uri` properties
2. Generate TechDocs
3. Feedback buttons are missing
4. Add `repo_url` property
5. Feedback buttons are shown

## Context

This is the `catalog-info.yaml` content used fo the component:

```
apiVersion: backstage.io/v1alpha1
kind: Component
metadata:
  name: developer-docs
  annotations:
    alliander.com/product-owner: some-address@domain.com
    backstage.io/techdocs-ref: dir:.
spec:
  type: documentation
  lifecycle: experimental
```

This is the `mkdocs.yml`:

```
site_name: "Developer Docs"

# repo_url: https://github.com/our-org/developer-docs <= buttons are shown when this line is added

nav:
  - Welkom: index.md
  - Help mee: help-mee.md
  .... more entries

plugins:
  - techdocs-core
``` 

## Your Environment

- Browser Information: 

- Output of `yarn backstage-cli info`: 

```text
OS:   Windows_NT 10.0.19044 - win32/x64
node: v14.17.6
yarn: 1.22.11
cli:  0.17.0 (installed)

Dependencies:
  @backstage/app-defaults                               1.0.1
  @backstage/backend-common                             0.13.2
  @backstage/backend-tasks                              0.3.0
  @backstage/catalog-client                             0.9.0, 1.0.1
  @backstage/catalog-model                              0.13.0, 1.0.1
  @backstage/cli-common                                 0.1.8
  @backstage/cli                                        0.17.0
  @backstage/config-loader                              1.1.0
  @backstage/config                                     0.1.15, 1.0.0
  @backstage/core-app-api                               1.0.1
  @backstage/core-components                            0.9.3
  @backstage/core-plugin-api                            0.8.0, 1.0.1
  @backstage/dev-utils                                  1.0.1
  @backstage/errors                                     0.2.2, 1.0.0
  @backstage/integration-react                          1.0.1
  @backstage/integration                                0.8.0, 1.1.0
  @backstage/plugin-api-docs                            0.8.4
  @backstage/plugin-app-backend                         0.3.31
  @backstage/plugin-auth-backend                        0.13.0
  @backstage/plugin-auth-node                           0.2.0
  @backstage/plugin-catalog-backend-module-github       0.1.2
  @backstage/plugin-catalog-backend                     1.1.1
  @backstage/plugin-catalog-common                      1.0.1
  @backstage/plugin-catalog-graph                       0.2.16
  @backstage/plugin-catalog-graphql                     0.3.8
  @backstage/plugin-catalog-import                      0.8.7
  @backstage/plugin-catalog-react                       0.9.0, 1.0.1
  @backstage/plugin-catalog                             1.1.0
  @backstage/plugin-explore-react                       0.0.16
  @backstage/plugin-explore                             0.3.35
  @backstage/plugin-github-actions                      0.5.4
  @backstage/plugin-graphql-backend                     0.1.21
  @backstage/plugin-home                                0.4.20
  @backstage/plugin-org                                 0.5.4
  @backstage/plugin-permission-common                   0.5.3, 0.6.0
  @backstage/plugin-permission-node                     0.6.0
  @backstage/plugin-permission-react                    0.3.4, 0.4.0
  @backstage/plugin-proxy-backend                       0.2.25
  @backstage/plugin-scaffolder-backend                  1.1.0
  @backstage/plugin-scaffolder-common                   1.0.1
  @backstage/plugin-scaffolder                          1.1.0
  @backstage/plugin-search-backend-module-pg            0.3.2
  @backstage/plugin-search-backend-node                 0.6.0
  @backstage/plugin-search-backend                      0.5.1
  @backstage/plugin-search-common                       0.3.3
  @backstage/plugin-search-react                        0.1.0
  @backstage/plugin-search                              0.8.0
  @backstage/plugin-sonarqube                           0.3.4
  @backstage/plugin-stack-overflow                      0.1.0
  @backstage/plugin-tech-insights-backend-module-jsonfc 0.1.15
  @backstage/plugin-tech-insights-backend               0.3.0
  @backstage/plugin-tech-insights-common                0.2.4
  @backstage/plugin-tech-insights-node                  0.2.9
  @backstage/plugin-tech-radar                          0.5.11
  @backstage/plugin-techdocs-backend                    1.1.0
  @backstage/plugin-techdocs-node                       0.11.12, 1.1.0
  @backstage/plugin-techdocs-react                      0.1.0
  @backstage/plugin-techdocs                            1.1.0
  @backstage/plugin-user-settings                       0.4.3
  @backstage/release-manifests                          0.0.2
  @backstage/search-common                              0.3.3
  @backstage/test-utils                                 1.0.1
  @backstage/theme                                      0.2.15
  @backstage/types                                      0.1.3, 1.0.0
  @backstage/version-bridge                             0.1.2, 1.0.1
```

**Comment by garyniemen** [comment]
This works on our instance internally. Any thoughts what is happening here @ottosichert ?

**Comment by leon-vg** [comment]
Not sure whether this helps, but in the demo the feedback buttons are also missing. Not sure whether this is by purpose, though: https://demo.backstage.io/docs/default/component/backstage.

**Comment by ottosichert** [comment]
Hey @leon-vg! Thanks for reporting the issue. As @garyniemen mentioned, we have already implemented this in our internal Backstage instance.

However we are using it as a newly introduced TechDocs Addon (see https://github.com/backstage/backstage/issues/9636) to provide additional information when creating a new issue.

This means we will release this functionality back in form of an addon instead of using the MkDocs generated link. We are currently finalizing the Addon Framework API and will continue to release individual addons over the next weeks

**Comment by benlangfeld** [comment]
@ottosichert This used to work. Was the functionality intentionally removed or does this count as a bug? If it was intentionally removed, why was that done before the substitute was available?

**Comment by garyniemen** [comment]
@camilaibs Can you take a look at this. The edit icon works internally. But in the open not sure. Also, there is a feedback button in the screenshot provided above. Do you know anything about that?

**Comment by dhhuynh2** [comment]
> Stepping through the code using breakpoints, I found that the [getRepoUrlFromLocationAnnotation](https://github.com/backstage/backstage/blob/c938e0d61dbe74f126e5e347413addba8bcd3698/plugins/techdocs-node/src/stages/generate/helpers.ts#L94) helper function only adds the repo_url OR edit_uri properties. It seems that to show the feedback buttons both properties are necessary (?).

I ran into the same problem as @leon-vg and dug into it a bit. Internally, changing this line from:

https://github.com/backstage/backstage/blob/c938e0d61dbe74f126e5e347413addba8bcd3698/plugins/techdocs-node/src/stages/generate/helpers.ts#L117

To:

`return { repo_url: target, edit_uri: integration.resolveEditUrl(sourceFolder) };`

Seems to work fine for our purposes, setting both `repo_url` and `edit_uri`. [getRepoUrlFromLocationAnnotation](https://github.com/backstage/backstage/blob/c938e0d61dbe74f126e5e347413addba8bcd3698/plugins/techdocs-node/src/stages/generate/helpers.ts#L94) seems to only be called from [mkDocsPatchers](https://github.com/backstage/backstage/blob/c938e0d61dbe74f126e5e347413addba8bcd3698/plugins/techdocs-node/src/stages/generate/mkDocsPatchers.ts#L104), so it seems like a safe change, unless I'm missing some context. I could get a PR opened quickly if that's the case.

`getRepoUrlFromLocationAnnotation` hasn't been changed in quite a while though, so I don't think it's responsible for the git feedback buttons disappearing. Can anyone advise?

**Comment by garyniemen** [comment]
@camilaibs Can you follow up on this.

**Comment by iamEAP** [comment]
> Seems to work fine for our purposes, setting both `repo_url` and `edit_uri`. [getRepoUrlFromLocationAnnotation](https://github.com/backstage/backstage/blob/c938e0d61dbe74f126e5e347413addba8bcd3698/plugins/techdocs-node/src/stages/generate/helpers.ts#L94) seems to only be called from [mkDocsPatchers](https://github.com/backstage/backstage/blob/c938e0d61dbe74f126e5e347413addba8bcd3698/plugins/techdocs-node/src/stages/generate/mkDocsPatchers.ts#L104), so it seems like a safe change, unless I'm missing some context. I could get a PR opened quickly if that's the case.

Hey @dhhuynh2, I don't see any issue with what you've proposed.  Would love a PR, if you're able to contribute that back!

**Comment by dhhuynh2** [comment]
I think this should be fixed on plugin-techdocs 1.3.0.

**Comment by leon-vg** [comment]
It looks like the edit button is back. The feedback button is not there, but has been replaced with the option for feedback when selecting a portion of text, right?

Closing this issue. Thanks for fixing this!

![image](https://user-images.githubusercontent.com/14820561/197703865-9f6ed509-2e8a-4592-9e53-c79380666da5.png)
